Description: Prevent composition if src outside clip region
 This patch prevents the image compositing occurring if the src is
 outside of the clip region.
Origin: http://git.ghostscript.com/?p=ghostpdl.git;a=commit;h=8b1505
Author: Shailesh Mistry <shailesh.mistry@hotmail.co.uk>
Bug: http://bugs.ghostscript.com/show_bug.cgi?id=690870
Forwarded: no
Last-Update: 2012-02-10

--- a/jbig2_image.c
+++ b/jbig2_image.c
@@ -198,7 +198,7 @@
     w = src->width;
     h = src->height;
     ss = src->data;
-    /* FIXME: this isn't sufficient for the < 0 cases */
+
     if (x < 0) { w += x; x = 0; }
     if (y < 0) { h += y; y = 0; }
     w = (x + w < dst->width) ? w : dst->width - x;
@@ -209,6 +209,15 @@
         w, h, x, y);
 #endif
 
+    /* check for zero clipping region */
+    if ((w <= 0) || (h <= 0))
+    {
+#ifdef JBIG2_DEBUG
+        jbig2_error(ctx, JBIG2_SEVERITY_DEBUG, -1, "zero clipping region");
+#endif
+        return 0;
+    }
+
 #if 0
     /* special case complete/strip replacement */
     /* disabled because it's only safe to do when the destination
